lib/pbench/server/api/__init__.py

```diff
@@ -272,21 +274,48 @@ class Upload(Resource):
     def post(self):
         global app
         if not request.headers.get("filename"):
-            app.logger.debug("Missed filename in header")
+            app.logger.debug("Tarfile upload: Missed filename in header")
             abort(400, message="Missing filename header in request")
         filename = secure_filename(request.headers.get("filename"))
 
         if not request.headers.get("md5sum"):
-            app.logger.debug("Missed md5sum in header")
+            app.logger.debug("Tarfile upload: Missed md5sum in header")
             abort(400, message="Missing md5sum header in request")
         md5sum = request.headers.get("md5sum")
 
         app.logger.debug("Receiving file: {}", filename)
         if not allowed_file(filename):
-            app.logger.debug("Bad file extension received")
+            app.logger.debug("Tarfile upload: Bad file extension received")
             abort(400, message="File extension not supported. Only .xz")
 
+        try:
+            content_length = int(request.headers.get("Content-Length"))
+        except ValueError:
+            app.logger.debug(
+                "Tarfile upload: Invalid content-length header, not an integer"
+            )
+            abort(400, message="Invalid content-length header, not an integer")
+        except Exception:
+            app.logger.debug("Tarfile upload: No Content-Length header value found")
+            abort(400, message="Missing required content-length header")
+        else:
+            if content_length > _MAX_UPLOAD_DATA_SIZE:
+                app.logger.debug(
+                    "Tarfile upload: Content-Length exceeded maximum upload allowed size"
+                )
+                abort(
+                    400,
+                    message=f"Content object too large, keep it at {humanize.naturalsize(_MAX_UPLOAD_DATA_SIZE)}"
+                    f" ({content_length:d}) and under",
+                )
+            elif content_length == 0:
+                app.logger.debug("Tarfile upload: Content-Length header value is 0")
+                abort(
+                    400, message="Payload Content-Length empty, nothing was uploaded",
```

Comment:
Again, this isn't really true; "nothing was uploaded" implies that we think they didn't upload data in the payload, but at this point we're only examining the header; this isn't going to help them figure out what they got wrong. The debug log message is actually more useful than the HTTP response.